doc/modules/ROOT/pages/design/middleware.adoc

```diff
@@ -154,3 +154,15 @@ fundamental to allowing simultaneous activities in the same nREPL.
 For instance - if you want to evaluate two expressions simultaneously
 you'll have to do this in separate session, as all requests within the
 same session are serialized.
+
+== Pretty Printing
+
+nREPL includes a `pr-values` middleware to render the results of evaluated
+forms as strings for returning to the client. By default, this will use either
+`print-dup` or `print-method` to match the standard Clojure `print` behavior.
+To customize this you can pass a custom `:renderer` as a symbol along with the
+message. If present, this will be resolved to a function and used to render the
+value instead.
+
+This enables using libraries like link:https://github.com/greglook/puget[puget]
+to pretty-print the evaluation results automatically.
```

Comment:
We might add here some example eval request using either Clojure's built-in pprint or puget so tools writers will have a better idea what they are supposed to do. I guess it's not immediately obvious that we'll pull out the `:renderer` from an `eval` op request.